src/Shared/HttpSys/NativeInterop/SocketAddress.cs

```diff
@@ -65,32 +65,21 @@ private IPAddress GetIPv4Address()
 
     internal static unsafe SocketAddress? CopyOutAddress(IntPtr address)
```

Comment:
Can the signature be changed to pass in the buffer size (ideally as Span)? This method won't need to be unsafe once you do that. 

Passing buffers without size is against secure coding best practices.

Reply:
Ultimately I can't hide the fact that the address is coming from Win32 as a `PSOCKADDR`, and the code needs to read the header to determine if this is really a `PSOCKADDR_IN` or a `PSOCKADDR_IN6`, which have different lengths.

I wonder if that means that the most straightforward thing to do would be to change the `IntPtr` argument to a `SOCKADDR*` argument. Yes, it means we're still dealing with pointers, but now at least all accesses are through an actual data type rather than magic offset values. Would probably help make the code more understandable.

Reply:
Yes, it would definitely look better.